### PR TITLE
Issue-12 - Add support for uploading directly to an NG server

### PR DESCRIPTION
**Pull request by TheRealDougSchmidt**
This change adds the ability to upload an EHSN file directly to an NG server.

It uses the AQTS python wrapper class to determine if the selected server is a 3.X server or an NG server.

If a 3.X server is detected, the same upload logic applies.
If an NG server is detected, the eHSN XML file and its PDF summary will be uploaded.

@DougStiff @AnojanS I've been bench-testing this change, so that I can upload both the eHSN XML file and its summary PDF into an AQUARIUS NG server. It seems to be working for me, but it definitely needs more testing that none of the existing 3.x workflows are broken. (I can't easily test the exist 3.X back end logic without access to a few more internal WSC systems)

The basic change is the addition of our [Python REST API wrapper class](https://github.com/AquaticInformatics/examples/tree/master/TimeSeries/PublicApis/Python) anda bit of glue logic.

The wrapper can quickly determine the version of the AQTS server, so when the [config.xml](https://github.com/ECCC-MSC/WSC-EHSN/blob/master/config%20-%20Template.xml) setting points to an NG server, the NG upload logic is used. Otherwise, it should fall back to the 3.X upload logic.

### CODE REVIEW

File: 1_ElectronicFieldNotes.py [hunk 1]
```diff
@@ -333,6 +335,49 @@ def CheckFVVals(self):
...
+                # Upload the ZIP archive
+                with open(zipPath, 'rb') as zipStream:
+                    response = timeseries.acquisition.post("/locations/" + locationUniqueId + "/visits/upload/plugins",
+                                         files={'file': zipStream}).json()
+
+                return True, None
+
+            except requests.exceptions.HTTPError as e:
+
+                return True, str(e) + "\n\nTo show this HTTPError, I had to add 'password' to the response value."
```
Comment (TheRealDougSchmidt) [inline 1_ElectronicFieldNotes.py:372]: This hacky bit of code is the main reason I wouldn't merge the code directly.

When an EHSN upload fails, the server responds with a helpful error message that should be shown to the user.

Some common error messages include:
`Saving parsed data would result in duplicates.` - You've already uploaded the file.
`ReferencePoint with name '{name}' for location {id} was not found` - The eHSN file contains level survey activity for benchmarks that don't exist in AQTS. Better add the benchmark and try again.

In order to show this error message, I had to hack in the phrase "password" somewhere in the text. If the response doesn't include the string "password", the existing eHSN upload logic assumes everything is successful. That's quite a logic bug that needs to be fixed, so I hacked in this phrase so that at least the error gets shown.

image
image

You'll really need to fix the error handling logic throughout the code base to make the eHSN app more robust.
